tensorflow/python/ops/clip_ops.py

```diff
@@ -243,13 +244,16 @@ def clip_by_global_norm(t_list, clip_norm, use_norm=None, name=None):
 
   Raises:
     TypeError: If `t_list` is not a sequence.
+    InvalidArgumentError: if `use_norm` is not finite.
   """
   if (not isinstance(t_list, collections.Sequence)
       or isinstance(t_list, six.string_types)):
     raise TypeError("t_list should be a sequence")
   t_list = list(t_list)
   if use_norm is None:
     use_norm = global_norm(t_list, name)
+  use_norm = numerics.verify_tensor_all_finite(use_norm,
+                                               "use_norm must be finite.")
```

Comment:
Something like "infinite global norm detected" seems to be a better error message since the user might not provide the `use_norm` argument to the function. In this case, the current message would be confusing,